packages/extension-api/src/extension-api.d.ts

```diff
@@ -1339,6 +1346,31 @@ declare module '@podman-desktop/api' {
     readonly location: Uri;
   }
 
+  export type KubernetesGeneratorSelector = KubernetesGeneratorType | ReadonlyArray<KubernetesGeneratorType>;
+
+  export type KubernetesGeneratorType = 'Compose' | 'Pod' | 'Container';
+
+  /**
+   * The result containing a Kubernetes config files
+   */
+  export interface GenerateKubeResult {
+    yaml: string;
+  }
+
+  /**
+   * The KubernetesGeneratorProvider allows an extension to register a custom Kube Generator for a specific
+   * KubernetesGeneratorType.
+   */
+  export interface KubernetesGeneratorProvider {
+    /**
+     * A unique identifier for the provider.
+     */
+    id: string;
+    name: string;
+    types: KubernetesGeneratorSelector;
+    generate(engineId: string, ids: string[]): GenerateKubeResult;
```

Comment:
based on your previous feedback, we need to change string[] to a set of resources like

```ts
{containers: [], pods: [], compose: []}
```

Reply:
What type for containers pods and compose ? If we use `ContainerInfoUI` we need to add them to the `extension-api.d.ts` ?

Reply:
just IDs so string

Reply:
composeInfoUI does not have a unique identifier

Reply:
it looks like we submit ids of containers when we're using the generate for compose usecase

Reply:
but we could specify 'com.docker.compose.project' label value ?

Reply:
> but we could specify 'com.docker.compose.project' label value ?

Humm, I think if we want to use that, we should not be able to send an array of it, I do think the generate function should support an array of compose project.

Reply:
I will use the containers id for now, as it is in the current implementation